front/lib/plans/subscription.ts

```diff
@@ -296,16 +299,42 @@ export const pokeUpgradeWorkspaceToPlan = async (
     );
   }
 
-  if (newPlan.code === PRO_PLAN_SEAT_29_CODE) {
+  const isUgradeToEnterprise = newPlan.code.startsWith("ENT_");
+  const isUpgradeToPro = isProPlanCode(newPlan.code);
+  const isCurrentlyPro =
+    activeSubscription && isProPlanCode(activeSubscription.plan.code);
+
+  // Ugrade to Enterprise is not allowed through this function.
+  if (isUgradeToEnterprise) {
     throw new Error(
-      `Cannot subscribe to plan ${planCode}: it is the pro plan which requires a stripe subscription.`
+      `Cannot subscribe to plan ${planCode}: Enterprise Plans requires a special process.`
+    );
+  }
+
+  // Upgrade to Pro is allowed only if the workspace is already subscribed to a Pro plan.
+  // This is a way to change the plan limitations but stay on Pro.
+  if (isUpgradeToPro) {
+    if (!isCurrentlyPro) {
+      throw new Error(
+        `Cannot subscribe to ${planCode}: Pro Plans requires a stripe checkout session done by the user on the product.`
+      );
+    }
+    await Subscription.update(
+      { planId: newPlan.id },
+      {
+        where: {
+          stripeSubscriptionId: activeSubscription.stripeSubscriptionId,
```

Comment:
Shouldn't we safeguard that by adding workspaceId in the where clause too?
Just in case we sometime in the future have something wonky in the db, making sure we're editing the right workspace not only based on the stripe subscription id

Reply:
I think we can trust that the stripeSubscriptionId that we fetched from the auth object from the workspace is attached to the valid workspace! That being said, I added a commit to base my update on the `sId` instead. Since it's unique, I'm sure I'm editing the exact same subscription object that I got from the authenticator.  👍🏻